### ISSUE TITLE
AttributeError: nominalWidthX with CFF2 font

### ISSUE THREAD

**Issue by miguelsousa** [fonttools/fonttools#1331]
The code snippet below works fine with a CFF font but throws an `AttributeError` with a CFF2 font.
It's not clear to me in which part of fontTools the fix should be made, given the current cff/cff2 architecture.

```python
import sys
from fontTools.ttLib import TTFont
from fontTools.pens.areaPen import AreaPen

font = TTFont(sys.argv[1])
glyph_set = font.getGlyphSet()
pen = AreaPen()
glyph_set['A'].draw(pen)
print(pen.value)
```

```python
Traceback (most recent call last):
  File "draw_cff2.py", line 8, in <module>
    glyph_set['A'].draw(pen)
  File "/fontTools/ttLib/ttFont.py", line 756, in draw
    self._glyph.draw(pen)
  File "/fontTools/misc/psCharStrings.py", line 982, in draw
    self.private.nominalWidthX, self.private.defaultWidthX)
  File "/fontTools/cffLib/__init__.py", line 2254, in __getattr__
    raise AttributeError(name)
AttributeError: nominalWidthX
```

**Comment by readroberts** [comment]
@miguelsousa I actually ran into this just yesterday. The psCharStrings .CFF2Subr class inherits the psCharStrings .T2CharString.draw() method, which assumes that private.nominalWidthX and self.private.defaultWidthX exist - and for CFF2Subr's, they don't.   Just to keep going with my own work, I made the following quick fix to the CFF2Subr class:
	def draw(self, pen):
		subrs = getattr(self.private, "Subrs", [])
		extractor = self.outlineExtractor(pen, subrs, self.globalSubrs,
				0, 0)
		extractor.execute(self)
		self.width = 0

but haven't yet looked to see if it makes more sense to make a new CFF22OutlineExtractor subclass that doesn't  require the nominalWidthX and  defaultWidthX values, and if I can get away with not setting self.width. I will revisit this in a day or so, when I finish my update to buildcff2vf.
